native/cocos/core/scene-graph/Node.cpp

```diff
@@ -550,16 +546,16 @@ void Node::updateWorldTransform() { //NOLINT(misc-no-recursion)
         auto *currChild = child;
         if (curr) {
             if (dirtyBits & static_cast<uint32_t>(TransformBit::POSITION)) {
-                currChild->_worldPosition.transformMat4(currChild->_localPosition, curr->getWorldMatrix());
+                currChild->_worldPosition.transformMat4(currChild->_localPosition, curr->_worldMatrix);
```

Comment:
currChild is child, curr is parent. parent's world transform should have already been updated. see ts logic:
```ts
                if (dirtyBits & TransformBit.POSITION) {
                    Vec3.transformMat4(child._pos, child._lpos, cur._mat); // cur._mat is private property.
                    child._mat.m12 = child._pos.x;
                    child._mat.m13 = child._pos.y;
                    child._mat.m14 = child._pos.z;
                }
```